### PR TITLE
'conan inspect' command and 'graph.load_conanfile_class' API

### PR DESCRIPTION
**Pull request by lasote**
close https://github.com/conan-io/conan/issues/11012

`conan inspect path <path>` => With no attribute selection.

### CODE REVIEW

File: conans/cli/commands/inspect.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    return json.dumps(data, indent=4)
+
+
+@conan_command(group=COMMAND_GROUPS['creator'], formatters={"json": _inspect_json_formatter})
+def inspect(conan_api, parser, *args, **kwargs):
+    """
+    Returns the specified attribute/s of a conanfile
+    """
+    parser.add_argument("path", help="Path to a folder containing a recipe (conanfile.py)")
+    parser.add_argument("attribute", help="Attribute of the conanfile to read", nargs="+")
```
Comment (memsharded) [inline conans/cli/commands/inspect.py]: Probably try to align this ``attribute`` with the ``--filter`` argument in ``conan graph info``

File: conans/cli/commands/inspect.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+def _inspect_json_formatter(data):
+    return json.dumps(data, indent=4)
+
+
+@conan_command(group=COMMAND_GROUPS['creator'], formatters={"json": _inspect_json_formatter})
+def inspect(conan_api, parser, *args, **kwargs):
+    """
+    Returns the specified attribute/s of a conanfile
+    """
+    parser.add_argument("path", help="Path to a folder containing a recipe (conanfile.py)")
```
Comment (memsharded) [inline conans/cli/commands/inspect.py]: Only for local conanfiles in user folder, so no possible to inspect one recipe in the cache? Will that need a custom command?

File: conans/cli/commands/inspect.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+from conans.cli.command import conan_command, COMMAND_GROUPS
+from conans.cli.output import ConanOutput
+from conans.errors import ConanException
+
+
+def _inspect_json_formatter(data):
+    return json.dumps(data, indent=4)
+
+
+@conan_command(group=COMMAND_GROUPS['creator'], formatters={"json": _inspect_json_formatter})
```
Comment (memsharded) [inline conans/cli/commands/inspect.py]: Maybe the inspect is useful as consumer, to know for example which options does a recipe have?

File: conans/cli/commands/inspect.py (hunk 2 above)
Comment (lasote) [reply conans/cli/commands/inspect.py]: `conan cache path` that's what we talked

File: conans/cli/commands/inspect.py (hunk 1 above)
Comment (lasote) [reply conans/cli/commands/inspect.py]: But a filter is to "discard all" and show only some fields. I'm not sure here we can/want to show a full set of fields from the recipe. Or do we? Which ones?

Comment (memsharded) [reply conans/cli/commands/inspect.py]: The original ``conan inspect`` listed many fields from the recipe, by default. I think I'd try to align on ``conan graph info`` output?

File: conans/cli/commands/inspect.py (hunk 2 above)
Comment (lasote) [reply conans/cli/commands/inspect.py]: Alternatives?

`conan inspect [--path] [--reference] attr1 attr2`  => bit messy

`conan inspect path attr1 attr2` + `conan inspect reference attr1 attr2` => a bit more clean

Do we see `conan inspect` as a `name` that we want to reserve for more use cases like packages?

**Review by czoido — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Review by memsharded — APPROVED** [review]

**Pull request merged**